cmd/set/puppet.go

```diff
@@ -0,0 +1,48 @@
+package set
+
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver"
+	"github.com/puppetlabs/prm/pkg/prm"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var PuppetSemVer *semver.Version
+
+func createSetPuppetCommand() *cobra.Command {
+	tmp := &cobra.Command{
+		Use:     "puppet <VERSION>",
+		Short:   "Sets the Puppet runtime to the specified version",
+		Long:    `Sets the Puppet runtime to the specified version`,
+		PreRunE: preExecute,
+		Run:     setPuppetVersion,
+	}
+
+	return tmp
+}
+
+func preExecute(cmd *cobra.Command, args []string) (err error) {
+	if len(args) > 1 {
+		return fmt.Errorf("too many args, please only specify a Puppet version after 'set puppet'")
+	}
+
+	if len(args) < 1 {
+		return fmt.Errorf("please specify a Puppet version after 'set puppet'")
+	}
+
+	PuppetSemVer, err = semver.NewVersion(args[0])
+	if err != nil {
+		return fmt.Errorf("'%s' is not a semantic Puppet version: %s", args[0], err)
+	}
```

Comment:
Move this into `setPuppetVersion` and use `RunE`. This will remove the need to store the semver in a global.